### PR TITLE
chore: info log whenever `ConfigGenApi::state` is changed

### LINKED ISSUE #3499: Whenever `ConfigGenApi::state` is changed, log this change at `INFO` level
This would help with debugging

### PR DESCRIPTION
**Pull request by mayrf**
Fixes #3499 

I am pretty sure I covered all occurrences of changes to `ConfigGenApi::state`, but I am not sure If `LOG_NET_PEER_DKG` is the appropriate log channel and I would appreciated suggestions for improving the messages.

### CODE REVIEW

File: fedimint-server/src/config/api.rs [hunk 1]
```diff
@@ -63,20 +63,26 @@ impl ConfigGenApi {
...
+        tracing::info!(target: fedimint_logging::LOG_NET_PEER_DKG, "Created new config gen Api");
+        config_gen_api
     }
 
     // Sets the auth and decryption key derived from the password
     pub fn set_password(&self, auth: ApiAuth) -> ApiResult<()> {
         let mut state = self.require_status(ServerStatus::AwaitingPassword)?;
         state.auth = Some(auth);
         state.status = ServerStatus::SharingConfigGenParams;
+        tracing::info!(
```
Comment (dpc) [inline fedimint-server/src/config/api.rs:82]: Fully qualifying logging statements seems unnecessary and verbose. We just do `info!` etc. everywhere.

**Review by dpc — COMMENTED** [review]
One style change, but LGTM otherwise.

File: fedimint-server/src/config/api.rs [hunk 2]
```diff
@@ -791,7 +822,7 @@ mod tests {
                     Ok(status) => return status,
                     Err(_) => sleep(Duration::from_millis(1000)).await,
                 }
-                tracing::info!(
+                info!(
```
Comment (elsirion) [inline fedimint-server/src/config/api.rs:825]: You also need to add `use tracing::info;` to the `tests` module.

**Review by dpc — APPROVED** [review]

**Comment by elsirion** [comment]
In the future you can also just [amend commits](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---amend) and force push or squash them in the end before merging, so there aren't any fixup commits for previous commits in the same PR.

**Pull request merged**